### PR TITLE
Support for nested data structures, function, and port in metadata

### PR DESCRIPTION
**Pull request by akasprzok**

### CODE REVIEW

**Comment by emoragaf** [comment]
Are we able to handle tuples now? (I didn't see a test for tuples)

File: test/logger_humio_backend/ingest_api/structured_test.exs [hunk 1]
```diff
@@ -113,7 +121,11 @@ defmodule Logger.Humio.Backend.IngestApi.StructuredTest do
                        "atom" => "gl",
                        "pid" => ^pid_string,
                        "reference" => ^reference_string,
-                       "string" => "some string"
+                       "string" => "some string",
+                       "map" => %{"map_key" => "map_value"},
+                       "list" => ["list_entry_1", "list_entry_2"],
```
Comment (emoragaf) [inline test/logger_humio_backend/ingest_api/structured_test.exs:126]: How are lists actually handled? I can't seem to find the `metadata/2` function for that case

**Comment by akasprzok** [comment]
> Are we able to handle tuples now? (I didn't see a test for tuples)

Not yet. I'm gonna pose the question for how to handle tuples in #observability.

File: test/logger_humio_backend/ingest_api/structured_test.exs (hunk 1 above)
Comment (akasprzok) [reply test/logger_humio_backend/ingest_api/structured_test.exs:126]: That's the nice thing; they aren't.

Iteraptor.map applies a function to all values inside a nested structure, so I can massage values that are nested.

The `Structured` IngestAPI can handle nested structures, so we don't actually need to flatten ourselves.

So now metadata.ex can simply focus on how to massage Port, Reference, etc. into a string without having to worry about the overall structure of the data.

**Review by emoragaf — APPROVED** [review]

**Pull request merged**